pkgs/development/python-modules/symengine/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, cython
+, cmake
+, symengine
+, nose
+}:
+
+buildPythonPackage rec {
+  pname = "symengine";
+  version = "0.3.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "e86d13aadc9f765f2c5462da32950edd36d1a0a52dbfc96e766be3689957c04d";
+  };
+
+  postConfigure = ''
+    substituteInPlace setup.py \
+      --replace "\"cmake\"" "\"${cmake}/bin/cmake\""
+
+    substituteInPlace cmake/FindCython.cmake \
+      --replace "SET(CYTHON_BIN cython" "SET(CYTHON_BIN ${cython}/bin/cython"
+  '';
+
+  buildInputs = [ cython cmake ];
+
+  setupPyBuildFlags = [ "--symengine-dir=${symengine}/" ];
+
+  # tests fail due to trying to import local "symengine" directory
```

Comment:
Ah but if you instead fetch the equivalent source from github:

```
  src = fetchFromGitHub {
    owner = "symengine";
    repo = "symengine.py";
    rev = "v0.3.0";
    sha256 = "1in32wr4dbb78zpi44j8dcldq368mkvi5l7hpw23v4llf2q7cx32";
  };
```

it gives you a script to invoke the tests, and following the way their `.travis.yml` does it, you can:

```
  checkInputs = [ pytest sympy ];
  checkPhase = ''
    mkdir empty
    cd empty
    ${python.interpreter} ../bin/test_python.py
  '';
```

et voila.